### PR TITLE
Adds method to retrieve a rollup by height.

### PR DESCRIPTION
**Pull request by joeldudley**
### Why is this change needed?

The API for `eth_getBalance` requires that we can check the balance at any block. Currently, there is no API to grab the rollup with a specific height.

### What changes were made as part of this PR:

Functional.

- Adds API to get rollup with a specific height

### CODE REVIEW

File: go/enclave/db/storage.go [hunk 1]
```diff
@@ -56,6 +56,19 @@ func (s *storageImpl) FetchGenesisRollup() *core.Rollup {
...
+func (s *storageImpl) FetchRollupByHeight(height uint64) *core.Rollup {
+	if height == 0 {
+		return s.FetchGenesisRollup()
+	}
+
+	hash := obscurorawdb.ReadCanonicalHash(s.db, height)
+	if hash == (gethcommon.Hash{}) {
+		return nil
+	}
+	r, _ := s.FetchRollup(hash)
```
Comment (BedrockSquirrel) [inline go/enclave/db/storage.go:68]: Should we also return boolean for success from this method?

**Review by BedrockSquirrel — COMMENTED** [review]
Minor thought that might help users of the code and seems consistent with other methods in storage code

File: go/enclave/db/storage.go (hunk 1 above)
Comment (joeldudley) [reply go/enclave/db/storage.go]: Yeah. Sorry, was copying `FetchGenesisRollup`, but that always succeeds.

**Review by BedrockSquirrel — APPROVED** [review]
LGTM

**Pull request merged**